lib/segment/src/index/field_index/numeric_index/immutable_numeric_index.rs

```diff
@@ -151,13 +150,19 @@ impl<T: Encodable + Numericable> NumericKeySortedVec<T> {
         }
     }
 
-    fn find_end_index(&self, bound: Bound<NumericIndexKey<T>>) -> usize {
+    fn find_end_index(&self, start: usize, bound: Bound<NumericIndexKey<T>>) -> usize {
+        if start >= self.data.len() {
+            return self.data.len();
```

Comment:
I guess start index has to be always not larger that end index. I propose to return `start` instead of `self.data.len()`
```suggestion
            return start;
```